async.js

```diff
@@ -24,32 +23,25 @@ var url = require('./url')
  */
 module.exports = function (size, callback) {
   size = size || 21
-  var buffer = Buffer.allocUnsafe(size)
-  if (callback) {
-    crypto.randomFill(buffer, 0, size, function (err, bytes) {
-      if (err) {
-        callback(err)
-      } else {
-        var id = ''
-        while (0 < size--) {
-          id += url[bytes[size] & 63]
-        }
-        callback(null, id)
+
+  var randomPromise = random.async(size)
+    .then(function (bytes) {
+      var id = ''
+      while (0 < size--) {
+        id += url[bytes[size] & 63]
       }
+      return id
     })
-  } else {
-    return new Promise(function (resolve, reject) {
-      crypto.randomFill(buffer, 0, size, function (err, bytes) {
-        if (err) {
-          reject(err)
-        } else {
-          var id = ''
-          while (0 < size--) {
-            id += url[bytes[size] & 63]
-          }
-          resolve(id)
-        }
-      })
-    })
+
+  if (!callback) {
+    return randomPromise
   }
+
+  randomPromise
```

Comment:
It will not work :D. `callback` API was created to support environments without `Promise`.

Reply:
Ok, I've got it. So I can't use Promise in `random.js` too :(